projects/graph/src/graph.py

```diff
@@ -6,4 +6,27 @@
 class Graph:
     """Represent a graph as a dictionary of vertices mapping labels to edges."""
     def __init__(self):
-        pass  # TODO
+        self.vertices = { }
+
+    def add_vertex(self, vertex):
+        self.vertices[vertex] = set()
+
+    def add_edge(self, vertex_a, vertex_b):
+        if vertex_a in self.vertices.keys() and vertex_b in self.vertices.keys():
+            self.vertices[vertex_a].add(vertex_b)
+            self.vertices[vertex_b].add(vertex_a)
+        else:
+            print("Cannot find all vertices")
```

Comment:
Great work Hunter, you're validating the vertices correctly but try to add some conditional logic to reference a specific vertex in the case where one of them is valid so you can point to the invalid one.